packages/example-codehub/src/codehub-application.ts

```diff
@@ -3,12 +3,46 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Application} from '@loopback/core';
+declare function require(name:string): any;
+const pinoHttp = require('pino-http');
+import {Application, Component, DefaultSequence, inject} from '@loopback/core';
+import {Provider} from '@loopback/context';
+import {BindingKeys, Logger, PinoLoggerComponent, WinstonLoggerComponent} from '@loopback/logger';
 import {UserController, HealthController} from './controllers';
 
+const loggerProviderKey = BindingKeys.System.LOGGER_PROVIDER;
+
+// compatible with PinoLoggerComponent
+class PinoHttpLoggerSequence extends DefaultSequence {
+  private defaultHandle: Function;
+  @inject(loggerProviderKey)
+  private loggerProvider: Provider<Logger>;
+  private httpLogger: any;
+  constructor(a: any, b: any, c: any, d: any) {
+    super(a, b, c, d);
+    this.defaultHandle = super.handle;
+  }
+  async handle(req: any, res: any) {
+    if (!this.httpLogger) {
+      this.httpLogger = pinoHttp({
```

Comment:
While this is functional, it's not following our design intentions for LoopBack next. See the newly added section https://github.com/strongloop/loopback-next/wiki/Writing-Components#modifying-request-handling-logic for more details.

IMO, the component should be exporting a provider for an action that can be used by user/application sequence. Something along the following lines:

```ts
import {SetupLogger} from '@loopback/logger';

class AppSequence extends DefaultSequence {
  constructor(
    @inject('sequence.actions.findRoute') protected findRoute: FindRoute,
    @inject('sequence.actions.invokeMethod') protected invoke: InvokeMethod,
    @inject('sequence.actions.send') public send: Send,
    @inject('sequence.actions.reject') public reject: Reject,
    @inject('logger.actions.setupLogger') public setupLogger: SetupLogger,
  ) {}

  async handle(req: ParsedRequest, res: ServerResponse) {
    this.setupLogger(req);
    await super(req, res);
  }
}
```